command/connect/envoy/exec_unix.go

```diff
@@ -0,0 +1,124 @@
+// +build linux darwin
+
+package envoy
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func execEnvoy(binary string, passThroughArgs []string, bootstrapJson []byte) error {
+	// Write the Envoy bootstrap config file out to disk in a pocket universe
+	// visible only to the current process (and exec'd future selves).
+	fd, err := writeEphemeralEnvoyTempFile(bootstrapJson)
+	if err != nil {
+		return errors.New("Could not write envoy bootstrap config to a temp file: " + err.Error())
+	}
+
+	// On unix systems after exec the file descriptors that we should see:
+	//
+	//   0: stdin
+	//   1: stdout
+	//   2: stderr
+	//   ... any open file descriptors from the parent without CLOEXEC set
+	//
+	// Above we explicitly disabled CLOEXEC for our temp file, so assuming
+	// FD numbers survive across execs, it should just be the value of
+	// `fd`. This is accessible as a file itself (trippy!) under
+	// /dev/fd/$FDNUMBER.
+	magicPath := filepath.Join("/dev/fd", strconv.Itoa(int(fd)))
+
+	// First argument needs to be the executable name.
+	envoyArgs := []string{
+		binary,
+		"--v2-config-only",
+		"--disable-hot-restart",
```

Comment:
I remembered to migrate this from my launch script. I don't think there's a bootstrap config setting to do this.

Reply:
Hmm I'm actually in two minds about how we should handle `--disable-hot-restart` now.

Ideally we want to still allow people to use this and hot-restart together if they want to.

We really only need this to allow for multiple envoys in the same cgroup for demo purposes - I wonder if it's better to just document the need to disable hot restart and have the user provide this manually if they want `-- --disable-hot-restart`? We already require them to override the admin server port if they want to run two instances in the same network namespace for testing...

Or perhaps nicer than that would be to do this by default but then. Sniff for a `RESTART_EPOCH` env var or `--restart-epoch` in the pass-through args and if either are set at all then leave this out?

Reply:
Oh the `RESTART_EPOCH` env var is only in the wrapper script - it expects to be passed to envoy via `--restart-epoch` so yeah maybe only include the disable IFF `--restart-epoch` doesn't exist in the other args? I'm writing up docs now and I'll document it as working this way but feel free to disagree and I can revise if you have a strong opinion on that.

Reply:
Leaving this for Paul to change post-merge as necessary.